File: .claude/hooks/frontend-lint-format.py
#!/usr/bin/env python
"""PostToolUse hook: prettier + eslint on edited TS/TSX files."""
import json
import subprocess
import sys
from pathlib import Path

data = json.load(sys.stdin)
file_path = Path(data.get("tool_input", {}).get("file_path", ""))

if file_path.suffix not in {".ts", ".tsx"}:
    sys.exit(0)

frontend_dir = Path(__file__).parent.parent.parent / "frontend"

subprocess.run(
    ["bunx", "prettier", "--write", str(file_path)],
    cwd=frontend_dir,
    check=False,
)

result = subprocess.run(
    ["bunx", "eslint", "--fix", str(file_path)],
    cwd=frontend_dir,
    capture_output=True,
    text=True,
)
if result.returncode != 0:
    print(result.stderr or result.stdout, file=sys.stderr)
    sys.exit(1)

File: .claude/hooks/frontend-lint-format.js
#!/usr/bin/env node
/**
 * PostToolUse hook: prettier + eslint on edited TS/TSX files.
 *
 * Exit codes:
 *   0 = allow
 *   1 = eslint failed (shown to Claude)
 */

'use strict';

const path = require('path');
const { spawnSync } = require('child_process');

const frontendDir = path.join(__dirname, '..', '..', 'frontend');

let data = '';
process.stdin.setEncoding('utf8');
process.stdin.on('data', chunk => { data += chunk; });
process.stdin.on('end', () => {
  let input = {};
  try { input = JSON.parse(data); } catch {}

  const filePath = (input && input.tool_input && input.tool_input.file_path) || '';
  if (!/\.(ts|tsx)$/.test(filePath)) process.exit(0);

  spawnSync('bunx', ['prettier', '--write', filePath], {
    cwd: frontendDir,
    stdio: 'ignore',
  });

  const result = spawnSync('bunx', ['eslint', '--fix', filePath], {
    cwd: frontendDir,
    encoding: 'utf8',
  });

  if (result.status !== 0) {
    process.stderr.write(result.stderr || result.stdout || '');
    process.exit(1);
  }

  process.exit(0);
});
